--- a/arkcompiler/jsvm/build_jsvm_inter.sh
+++ b/arkcompiler/jsvm/build_jsvm_inter.sh
@@ -136,5 +136,8 @@ do_env() {
     argurment+=" -D__MUSL__  -D_LIBCPP_HAS_MUSL_LIBC -DOHOS_JS_ENGINE -DENABLE_HISYSEVENT"
-    argurment+=" -DV8_COMPRESS_POINTERS -DV8_COMPRESS_POINTERS_IN_SHARED_CAGE -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_HAVE_TARGET_OS -DV8_TARGET_OS_OHOS -DV8_ENABLE_DRUMBRAKE"
+    argurment+=" -DV8_HAVE_TARGET_OS -DV8_TARGET_OS_OHOS -DV8_ENABLE_DRUMBRAKE"
+    if [[ "${TARGET_CPU}" != "arm" ]]; then
+        argurment+=" -DV8_COMPRESS_POINTERS -DV8_COMPRESS_POINTERS_IN_SHARED_CAGE -DV8_31BIT_SMIS_ON_64BIT_ARCH"
+    fi
     argurment+=" -fstack-protector-strong"
     argurment+=" -Wl,-z,noexecstack"
     argurment+=" -Wl,-z,relro"
