diff --git a/arkcompiler/jsvm/src/dfx/parse_jitcode.cpp b/arkcompiler/jsvm/src/dfx/parse_jitcode.cpp
--- a/arkcompiler/jsvm/src/dfx/parse_jitcode.cpp
+++ b/arkcompiler/jsvm/src/dfx/parse_jitcode.cpp
@@ -95,7 +95,7 @@ JitSymbolVMA::JitSymbolVMA(uint32_t pid) : isCurrentProcess(static_cast<uint32_t
             return;
         }
 
-        if (!ReadProcessMemory(pid, reinterpret_cast<uint64_t>(startMapAddress), address, memorySize)) {
+        if (!ReadProcessMemory(pid, static_cast<uint64_t>(startMapAddress), address, memorySize)) {
             LOG(Error) << "failed read dfxAddress to local address.";
             munmap(address, memorySize);
             return;
