--- a/src/v8/BUILD.gn
+++ b/src/v8/BUILD.gn
@@ -8644,14 +8644,6 @@ action("copy_v8_include") {
   outputs = [ report_file ]
 }
 
-if (is_ohos) {
-  action("build_v8_script") {
-    script = "//arkweb/chromium_ext/v8/build_v8_script.py"
-    args = []
-    outputs = [ "$target_out_dir/build_v8_script.so" ]
-  }
-}
-
 v8_shared_library("v8_shared") {
   public_deps = [
     ":v8_base",
@@ -8661,10 +8653,6 @@ v8_shared_library("v8_shared") {
     ":v8_snapshot",
     ":copy_v8_include",
   ]
-  if (is_ohos) {
-    deps = [":build_v8_script"]
-  }
-
   public_configs = [ ":external_config" ]
 }
 
