diff --git a/src/components/BUILD.gn b/src/components/BUILD.gn
--- a/src/components/BUILD.gn
+++ b/src/components/BUILD.gn
@@ -980,7 +980,9 @@ test("components_unittests") {
   # extensions targets.
   assert_no_deps = []
   if (!arkweb_unittests) {
-    assert_no_deps += [ "//chrome/*" ]
+    if (!is_ohos) {
+      assert_no_deps += [ "//chrome/*" ]
+    }
     assert_no_deps += disallowed_extension_deps_
   }
 
