diff --git a/src/components/BUILD.gn b/src/components/BUILD.gn
--- a/src/components/BUILD.gn
+++ b/src/components/BUILD.gn
@@ -1311,9 +1311,9 @@ if (!is_ios) {
     # No components should depend on Chrome.
     # Since //chrome depends on //extensions, we also only assert_no_deps on
     # extensions targets.
-    if (!arkweb_unittests) {
-    assert_no_deps = [ "//chrome/*" ]
-    assert_no_deps += disallowed_extension_deps_
+    if (!arkweb_unittests && !is_ohos) {
+      assert_no_deps = [ "//chrome/*" ]
+      assert_no_deps += disallowed_extension_deps_
     }
 
     if (is_fuchsia) {
