diff --git a/drivers/peripheral/audio/audio.gni b/drivers/peripheral/audio/audio.gni
index 9c84646..4770bd4 100644
--- a/drivers/peripheral/audio/audio.gni
+++ b/drivers/peripheral/audio/audio.gni
@@ -26,6 +26,7 @@ declare_args() {
   drivers_peripheral_audio_feature_hisysevent_enable = true
   drivers_peripheral_audio_feature_frame_ec = false
   drivers_peripheral_audio_feature_reclaim_memory = false
+  drivers_peripheral_audio_vendor_alsa_path = ""
 }
 
 if (defined(ohos_lite)) {
diff --git a/drivers/peripheral/audio/hdi_service/supportlibs/BUILD.gn b/drivers/peripheral/audio/hdi_service/supportlibs/BUILD.gn
index 62e57c0..276fe7b 100644
--- a/drivers/peripheral/audio/hdi_service/supportlibs/BUILD.gn
+++ b/drivers/peripheral/audio/hdi_service/supportlibs/BUILD.gn
@@ -82,7 +82,13 @@ if (defined(ohos_lite)) {
         sources +=
             [ "$hdf_audio_path/supportlibs/alsa_adapter/src/vendor_capture.c" ]
       } else {
-        sources += [ "//device/board/${product_company}/${device_name}/audio_alsa/vendor_capture.c" ]
+        if (drivers_peripheral_audio_vendor_alsa_path != "") {
+          sources +=
+              [ "${drivers_peripheral_audio_vendor_alsa_path}/vendor_capture.c" ]
+        } else {
+          sources +=
+              [ "//device/board/${product_company}/${device_name}/audio_alsa/vendor_capture.c" ]
+        }
       }
     } else {
       sources += [
@@ -100,8 +106,11 @@ if (defined(ohos_lite)) {
     if (drivers_peripheral_audio_feature_alsa_lib) {
       include_dirs += [ "$hdf_audio_path/supportlibs/alsa_adapter/include" ]
       if (!defined(is_emulator) || !is_emulator) {
-        include_dirs +=
-            [ "//device/board/${product_company}/${device_name}/audio_alsa" ]
+        if (drivers_peripheral_audio_vendor_alsa_path != "") {
+          include_dirs += [ drivers_peripheral_audio_vendor_alsa_path ]
+        } else {
+          include_dirs += [ "//device/board/${product_company}/${device_name}/audio_alsa" ]
+        }
       }
     } else {
       include_dirs += [ "$hdf_audio_path/supportlibs/adm_adapter/include" ]
@@ -160,7 +169,13 @@ if (defined(ohos_lite)) {
         sources +=
             [ "$hdf_audio_path/supportlibs/alsa_adapter/src/vendor_render.c" ]
       } else {
-        sources += [ "//device/board/${product_company}/${device_name}/audio_alsa/vendor_render.c" ]
+        if (drivers_peripheral_audio_vendor_alsa_path != "") {
+          sources +=
+              [ "${drivers_peripheral_audio_vendor_alsa_path}/vendor_render.c" ]
+        } else {
+          sources +=
+              [ "//device/board/${product_company}/${device_name}/audio_alsa/vendor_render.c" ]
+        }
       }
     } else {
       sources += [
@@ -179,8 +194,11 @@ if (defined(ohos_lite)) {
     if (drivers_peripheral_audio_feature_alsa_lib) {
       include_dirs += [ "$hdf_audio_path/supportlibs/alsa_adapter/include" ]
       if (!defined(is_emulator) || !is_emulator) {
-        include_dirs +=
-            [ "//device/board/${product_company}/${device_name}/audio_alsa" ]
+        if (drivers_peripheral_audio_vendor_alsa_path != "") {
+          include_dirs += [ drivers_peripheral_audio_vendor_alsa_path ]
+        } else {
+          include_dirs += [ "//device/board/${product_company}/${device_name}/audio_alsa" ]
+        }
       }
     } else {
       include_dirs += [ "$hdf_audio_path/supportlibs/adm_adapter/include" ]
